libs/masset-provider/src/lib/MassetProvider.tsx

```diff
@@ -47,9 +47,12 @@ export const useSelectedMassetName = (): MassetName => useSelectedMassetNameCtx(
 
 export const useSetSelectedMassetName = (): Dispatch<SetStateAction<MassetName>> => useSelectedMassetNameCtx()[1]
 
-export const useSelectedMassetConfig = (): MassetConfig => useContext(massetConfigCtx)
+export const useSelectedMassetConfig = (): MassetConfig => {
+  const masset = useSelectedMassetName()
+  return MASSET_CONFIG[masset]
+}
 
 export const MassetProvider = composedComponent(SelectedMassetNameProvider, ({ children }) => {
-  // const massetName = useSelectedMassetName()
-  return <massetConfigCtx.Provider value={MASSET_CONFIG.musd}>{children}</massetConfigCtx.Provider>
+  const massetName = useSelectedMassetName()
+  return <massetConfigCtx.Provider value={MASSET_CONFIG[massetName]}>{children}</massetConfigCtx.Provider>
```

Comment:
Looks good but I wonder if there was a reason JL changed this [before](https://github.com/mstable/mStable-apps/commit/30875971024c4ce2aa3203c89dab67986c1d2daa#diff-5b289deb585bf2a72f5351e8b499da14ea29c1050b200870d201ba17bb6cc0d7L53-L54)

Reply:
Looks like it was a bug before, I don't see a reason in the history. Good find